drivers/ipvlan/ipvlan_network.go

```diff
@@ -76,22 +80,31 @@ func (d *driver) CreateNetwork(nid string, option map[string]interface{}, nInfo
 }
 
 // createNetwork is used by new network callbacks and persistent network cache
-func (d *driver) createNetwork(config *configuration) error {
+func (d *driver) createNetwork(config *configuration) (bool, error) {
+	foundExisting := false
 	networkList := d.getNetworks()
 	for _, nw := range networkList {
 		if config.Parent == nw.config.Parent {
-			return fmt.Errorf("network %s is already using parent interface %s",
-				getDummyName(stringid.TruncateID(nw.config.ID)), config.Parent)
+			if config.ID != nw.config.ID {
+				return false, fmt.Errorf("network %s is already using parent interface %s",
+					getDummyName(stringid.TruncateID(nw.config.ID)), config.Parent)
+			}
+			logrus.Debugf("Create Network for the same ID %s\n", config.ID)
+			foundExisting = true
+			break
 		}
 	}
 	if !parentExists(config.Parent) {
 		// if the --internal flag is set, create a dummy link
 		if config.Internal {
-			err := createDummyLink(config.Parent, getDummyName(stringid.TruncateID(config.ID)))
-			if err != nil {
-				return err
+			if !dummyLinkExists(getDummyName(stringid.TruncateID(config.ID))) {
+				err := createDummyLink(config.Parent, getDummyName(stringid.TruncateID(config.ID)))
+				if err != nil {
+					return false, err
+				}
+				config.CreatedSlaveLink = true
 			}
-			config.CreatedSlaveLink = true
+
```

Comment:
I see in macvlan code with an else condition with prints an log message saying "dummy link exists ", should  we have similar warning here ?

Reply:
added the logs